File: apps/cpc-platform/android/app/src/main/kotlin/com/cpc/social/models/Product.kt
package com.cpc.social.models

import com.cpc.social.models.Money
import com.cpc.social.models.WarehouseLocation

data class Product(
    val id: String,
    val name: String,
    val brand: String? = null,
    val description: String,
    val barcode: String? = null,
    val carbonFootprint: Double? = null,
    val packagingType: String? = null,
    val nutritionalInfo: String? = null,
    val manufacturer: String? = null,
    val materialCost: Double? = null,
    val laborCost: Double? = null,
    val supplier: String? = null,
    val currentStock: Int? = null,
    val reorderLevel: Int? = null,
    val supplyChain: String? = null,
    val cost: Money? = null,
    val location: WarehouseLocation? = null
)

File: apps/platform/src-tauri/src/api/product.rs
use crate::types::Product;
use crate::mock::mock_supply_chain;
use tauri::State;
use sqlx::PgPool;

#[tauri::command]
pub async fn get_product_by_id(
    id: String,
    pool: State<'_, PgPool>
) -> Result<Product, String> {
    // In development, return mock product data
    #[cfg(debug_assertions)]
    if id == "mock" {
        let mut mock_product = Product {
            id: "mock".to_string(),
            name: "Organic Coffee".to_string(),
            brand: "Ethical Beans Co.".to_string(),
            description: "Premium organic coffee beans".to_string(),
            barcode: "1234567890".to_string(),
            carbon_footprint: 2.5,
            packaging_type: "Compostable bag".to_string(),
            nutritional_info: "".to_string(),
            manufacturer: "Green Coffee Processors".to_string(),
            material_cost: 5.0,
            labor_cost: 3.0,
            supplier: "Global Distributors".to_string(),
            current_stock: 100,
            reorder_level: 50,
            // supply_chain: Paused feature development
            // supply_chain: Some(mock_supply_chain()),
        };
        return Ok(mock_product);
    }

    // TODO: Implement actual database lookup in production
    // TODO: Implement product database lookup
    // TODO: Implement supply chain when feature resumes
    Err("Product not found".to_string())
}